src/artrace.py

```diff
@@ -315,15 +315,24 @@ def trace_orders(slf, mstrace, det, pcadesc="", maskBadRows=False, singleSlit=Fa
     msgs.info("Detecting order edges")
     #debugger.set_trace()
     if singleSlit:
-        msgs.info("Detecting slit edges")
-        filt = ndimage.sobel(binarr, axis=1, mode='constant')
-        msgs.info("Applying bad pixel mask")
-        filt *= (1.0-binbpx)  # Apply to the old detection algorithm
-        amin = np.argmin(filt, axis=1)
-        amax = np.argmax(filt, axis=1)
         edgearr = np.zeros(binarr.shape, dtype=np.int)
-        edgearr[np.arange(edgearr.shape[0]), amin] = +1
-        edgearr[np.arange(edgearr.shape[0]), amax] = -1
+        detect = True
+        if len(slf._argflag['trace']['orders']['sng_slit']) > 0:
+            redge = (det-1)*2+1
+            if slf._argflag['trace']['orders']['sng_slit'][redge] > 0:
+                msgs.warn("Using input slit edges. Better know what you are doing..")
```

Comment:
It might be a good idea to document somewhere (even if it's just a comment line here or, preferably, print out a very brief description of the required format with this warning message) the input format that is required. I presume in this approach one needs to specify the slit edges for *all* detectors in the mosaic (even if the slit edges of just one detector is bad).

I'm about to give the slit edge traces a slight overhaul when I have a chance to make more progress on the armed (i.e. APF) pipeline, and I'll test if my new algorithm works well with LRISr...